### PR TITLE
[io] prevent nullptr access if oldclass not found

### PR DESCRIPTION
**Pull request by ferdymercury**
# This Pull request:

## Changes or fixes:

Fixes https://github.com/root-project/root/issues/8194

## Checklist:

- [x] tested changes locally
- [ ] updated the docs (if necessary)

### CODE REVIEW

**Comment by linev** [comment]
I will wait until @pcanal will review the PR.

**Comment by dpiparo** [comment]
Thanks a lot for proposing this solution, appreciated!

File: io/io/src/TStreamerInfoActions.cxx [hunk 1]
```diff
@@ -429,7 +429,7 @@ namespace TStreamerInfoActions
          TClass *oldClass = aElement->GetClassPointer();
          if( vers < 9 && newClass && newClass!=oldClass ) {
             Error( "ReadBuffer", "Unfortunately, version %d of TStreamerInfo (used in %s) did not record enough information to convert a %s into a %s.",
-                  vers, buf.GetParent() ? buf.GetParent()->GetName() : "memory/socket", oldClass->GetName(), newClass->GetName() );
+                  vers, buf.GetParent() ? buf.GetParent()->GetName() : "memory/socket", oldClass ? oldClass->GetName() : "null", newClass->GetName() );
```
Comment (pcanal) [inline io/io/src/TStreamerInfoActions.cxx:432]:
```suggestion
            Error( "ReadBuffer", "Unfortunately, version %d of TStreamerInfo (used in %s) did not record enough information to convert a %s%s into a %s.",
                  vers, buf.GetParent() ? buf.GetParent()->GetName() : "memory/socket", oldClass ? oldClass->GetName() : aElement->GetTypeName(), oldClass ? "" : " (could not find the corresponding TClass)", newClass->GetName() );
```

**Review by pcanal — APPROVED** [review]

File: io/io/src/TStreamerInfoActions.cxx [hunk 2]
```diff
@@ -1276,8 +1276,8 @@ namespace TStreamerInfoActions
 
       if( vers < 8 ) {
          Error( "ReadSTLMemberWiseChangedClass", "Unfortunately, version %d of TStreamerInfo (used in %s) did not record enough information to convert a %s into a %s.",
-               vers, buf.GetParent() ? buf.GetParent()->GetName() : "memory/socket", oldClass->GetName(), newClass->GetName() );
-      } else {
+               vers, buf.GetParent() ? buf.GetParent()->GetName() : "memory/socket", oldClass ? oldClass->GetName() : "null", newClass ? newClass->GetName() : "null" );
```
Comment (pcanal) [inline io/io/src/TStreamerInfoActions.cxx:1279]: See https://github.com/root-project/root/pull/15876/files#r1651649143

File: io/io/src/TStreamerInfoActions.cxx [hunk 3]
```diff
@@ -1320,8 +1320,8 @@ namespace TStreamerInfoActions
...
```
Comment (pcanal) [inline io/io/src/TStreamerInfoActions.cxx:1323]: See https://github.com/root-project/root/pull/15876/files#r1651649143

**Review by pcanal — APPROVED** [review]

**Comment by pcanal** [comment]
Thanks.

**Pull request merged**